kernel/kernel-api/src/test/scala/io/delta/kernel/internal/tablefeatures/TableFeaturesSuite.scala

```diff
@@ -18,18 +18,165 @@ package io.delta.kernel.internal.tablefeatures
+  /////////////////////////////////////////////////////////////////////////////////////////////////
+  val readerWriterFeatures = Seq("columnMapping", "deletionVectors", "timestampNtz",
+    "typeWidening", "typeWidening-preview", "v2Checkpoint", "vacuumProtocolCheck",
+    "variantType", "variantType-preview")
+
+  val writerOnlyFeatures = Seq("appendOnly", "invariants", "checkConstraints",
+    "generatedColumns", "changeDataFeed", "identityColumns",
+    "rowTracking", "domainMetadata", "icebergCompatV2", "inCommitTimestamp")
+
+  val legacyFeatures = Seq("appendOnly", "invariants", "checkConstraints",
+    "generatedColumns", "changeDataFeed", "identityColumns", "columnMapping")
+
+  test("basic properties checks") {
+
+    // Check that all features are correctly classified as reader-writer or writer-only
+    readerWriterFeatures.foreach { feature =>
+      assert(TableFeatures.getTableFeature(feature).isReaderWriterFeature)
+    }
+    writerOnlyFeatures.foreach { feature =>
+      assert(!TableFeatures.getTableFeature(feature).isReaderWriterFeature)
+    }
+
+    // Check that legacy features are correctly classified as legacy features
+    (readerWriterFeatures ++ writerOnlyFeatures) foreach { feature =>
+      if (legacyFeatures.contains(feature)) {
+        assert(TableFeatures.getTableFeature(feature).isLegacyFeature)
+      } else {
+        assert(!TableFeatures.getTableFeature(feature).isLegacyFeature)
+      }
+    }
+
+    // all expected features are present in list. Just make sure we don't miss any
+    // adding to the list. This is the list used to iterate over all features
+    assert(
+      TableFeatures.TABLE_FEATURES.size() == readerWriterFeatures.size + writerOnlyFeatures.size)
+  }
+
+  val testProtocol = new Protocol(1, 2, Collections.emptyList(), Collections.emptyList())
+  Seq(
+    // Test feature, metadata, expected result
+    ("appendOnly", testMetadata(tblProps = Map("delta.appendOnly" -> "true")), true),
+    ("appendOnly", testMetadata(tblProps = Map("delta.appendOnly" -> "false")), false),
+    ("invariants", testMetadata(includeInvaraint = true), true),
+    ("invariants", testMetadata(includeInvaraint = false), false),
+    ("checkConstraints", testMetadata(tblProps = Map("delta.constraints.a" -> "a = b")), true),
+    ("checkConstraints", testMetadata(), false),
+    ("generatedColumns", testMetadata(includeGeneratedColumn = true), true),
+    ("generatedColumns", testMetadata(includeGeneratedColumn = false), false),
+    ("changeDataFeed",
+      testMetadata(tblProps = Map("delta.enableChangeDataFeed" -> "true")), true),
+    ("changeDataFeed",
+      testMetadata(tblProps = Map("delta.enableChangeDataFeed" -> "false")), false),
+    ("identityColumns", testMetadata(includeIdentityColumn = true), true),
+    ("identityColumns", testMetadata(includeIdentityColumn = false), false),
+    ("columnMapping", testMetadata(tblProps = Map("delta.columnMapping.mode" -> "id")), true),
```

Comment:
Q: do we support `name` mode in Kernel yet?

Reply:
we do support. The test is testing one of them to make sure we are auto enabling the column mapping feature.